### ISSUE TITLE
AttributeError: 'SysLogHandler' object has no attribute 'socket' when npm-accel writes logs

### ISSUE THREAD

**Issue by blaise-io** [xolox/python-npm-accel#4]
I'm getting `AttributeError: 'SysLogHandler' object has no attribute 'socket'` when npm-accel writes logs.

I guess this is the same issue as https://github.com/xolox/python-rotate-backups/issues/9

```
>>> virtualenv env -p python3
created virtual environment CPython3.8.5.final.0-64 in 1596ms
  creator CPython3Posix(dest=/Users/blaise/env/test/env, clear=False, global=False)
  seeder FromAppData(download=False, pip=bundle, wheel=bundle, setuptools=bundle, via=copy, app_data_dir=/Users/blaise/Library/Application Support/virtualenv)
    added seed packages: pip==20.2.2, setuptools==49.1.0, wheel==0.35.1
  activators PythonActivator,FishActivator,XonshActivator,CShellActivator,PowerShellActivator,BashActivator

>>> source env/bin/activate

>>> pip install npm-accel
Collecting npm-accel
  Using cached npm_accel-2.0.1-py2.py3-none-any.whl (20 kB)
Collecting humanfriendly>=8.0
  Using cached humanfriendly-8.2-py2.py3-none-any.whl (86 kB)
Collecting chardet>=2.3.0
  Using cached chardet-3.0.4-py2.py3-none-any.whl (133 kB)
Collecting verboselogs>=1.5
  Using cached verboselogs-1.7-py2.py3-none-any.whl (11 kB)
Collecting executor>=18.0
  Using cached executor-23.1-py2.py3-none-any.whl (85 kB)
Collecting property-manager>=3.0
  Using cached property_manager-3.0-py2.py3-none-any.whl (23 kB)
Collecting six>=1.9.0
  Using cached six-1.15.0-py2.py3-none-any.whl (10 kB)
Collecting fasteners>=0.14.1
  Using cached fasteners-0.15-py2.py3-none-any.whl (23 kB)
Collecting coloredlogs>=3.5
  Using cached coloredlogs-14.0-py2.py3-none-any.whl (43 kB)
Collecting monotonic>=0.1
  Using cached monotonic-1.5-py2.py3-none-any.whl (5.3 kB)
Installing collected packages: humanfriendly, chardet, verboselogs, six, monotonic, fasteners, property-manager, coloredlogs, executor, npm-accel
Successfully installed chardet-3.0.4 coloredlogs-14.0 executor-23.1 fasteners-0.15 humanfriendly-8.2 monotonic-1.5 npm-accel-2.0.1 property-manager-3.0 six-1.15.0 verboselogs-1.7
WARNING: You are using pip version 20.2.2; however, version 20.2.4 is available.
You should consider upgrading via the '/Users/blaise/env/test/env/bin/python -m pip install --upgrade pip' command.

>>> npm-accel
2020-11-05 09:45:03 Blaises-MacBook-Air.local npm_accel[6798] WARNING No dependencies extracted from ~/env/test/package.json file?!
--- Logging error ---
Traceback (most recent call last):
  File "/usr/local/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/lib/python3.8/logging/handlers.py", line 934, in emit
    self.socket.send(msg)
AttributeError: 'SysLogHandler' object has no attribute 'socket'
Call stack:
  File "/Users/blaise/env/test/env/bin/npm-accel", line 8, in <module>
    sys.exit(main())
  File "/Users/blaise/env/test/env/lib/python3.8/site-packages/npm_accel/cli.py", line 204, in main
    method(directory)
  File "/Users/blaise/env/test/env/lib/python3.8/site-packages/npm_accel/__init__.py", line 454, in install
    dependencies = self.extract_dependencies(package_file)
  File "/Users/blaise/env/test/env/lib/python3.8/site-packages/npm_accel/__init__.py", line 382, in extract_dependencies
    logger.warning("No dependencies extracted from %s file?!", formatted_path)
Message: 'No dependencies extracted from %s file?!'
Arguments: ('~/env/test/package.json',)
```

**Comment by xolox** [comment]
Another colleague of mine (@youtux) pointed out https://bugs.python.org/issue38780 which definitely looks related, given that I know and can tell from the shell transcript above that @blaise-io is also using MacOS.

**Comment by xolox** [comment]
Thanks for the feedback Blaise! Today I published coloredlogs [release 15.0](https://coloredlogs.readthedocs.io/en/latest/changelog.html#release-15-0-2020-12-10) which resolves this issue on MacOS. For now the new release of coloredlogs isn't a strict dependency of npm-accel (yet), but I do consider this issue resolved (as a fresh install won't exhibit the problem and `pip install --upgrade coloredlogs` can be used to resolve the existing issue).
